### ISSUE TITLE
datalink - default network interface

### ISSUE THREAD

**Issue by noah04** [libpnet/libpnet#404]
**TL;DR: Is a function for getting the default network interface needed/wanted?**

I'm working on a project where I use `libpnet` for packet sniffing. While porting from `pcap` (https://github.com/ebfull/pcap) to `libpnet`, I was missing an implementation to get the default network interface (capable of capturing packets).

With some research I found out that `pcap` uses a call to the `pcap_lookupdev` function[<sup>1</sup>]. Looking at the C code, I saw that they deprecated it anyway and use the first interface capable available that is not loopback[<sup>2</sup>].

With this in mind, I implemented the following naive helper-function in my project that works well. Filtering for valid IPs was necessary for me but I don't know if this is the general case or only for sniffing.

```rust
/// Tries to find the best suitable candidate for the default network interface:
/// It should be the first interface that is up, not loopback and has an ip.
fn default_interface(interfaces: &Vec<NetworkInterface>) -> Option<&NetworkInterface> {
    interfaces
        .iter()
        .filter(|e| e.is_up() && !e.is_loopback() && e.ips.len() > 0)
        .next()
}
```

 I wondered if you may want/need/plan a helper like this in the library? If this is the case and if you tell me a good place for adding it, I'd add a PR.

---

1: https://github.com/ebfull/pcap/blob/master/src/lib.rs#L201
2: https://github.com/the-tcpdump-group/libpcap/blob/master/pcap/pcap.h#L372

[<sup>1</sup>]: https://github.com/ebfull/pcap/blob/master/src/lib.rs#L201
[<sup>2</sup>]: https://github.com/the-tcpdump-group/libpcap/blob/master/pcap/pcap.h#L372

**Comment by JuxhinDB** [comment]
HI @noah04 -- thanks for the detailed issue and research. We may want to implement this as a helper function. I think for the time being, we could just document this nicely in our docs so that it's still accessible and simple to implement. What do you think?

**Comment by noah04** [comment]
I created a PR where I wrote some documentation on this. 

I added it to the `pnet_datalink::interfaces` functions documentation, since it wouldn't really fit into the [initial example](https://docs.rs/pnet/0.25.0/pnet/#ethernet-echo-server) as there is command-line-parsing involved to choose an interface.

Please note that this PR adds a new doc-test. It's designed not to fail unless the function itself changes. A non-existing default interface is handled. The test passes on my machine.

I saw that the other doc-test in `pnet_datalink/lib.rs` is set to be ignored, so if you want me to tag mine as ignored as well, please give me a hint.

**Comment by JuxhinDB** [comment]
Hey @noah04 -- I replied on your PR. Went ahead and merged it, for the time being, this is fine (let's not ignore the doc-test. Thanks again for the help on this. :-)
